Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,578 @@
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity: 10];
+  return array;
+}
+
+// Designated initializer...
++ (instancetype) constraintWithItem: (id)view1 
+                          attribute: (NSLayoutAttribute)attr1 
+                          relatedBy: (NSLayoutRelation)relation 
+                             toItem: (id)view2 
+                          attribute: (NSLayoutAttribute)attr2 
+                         multiplier: (CGFloat)mult 
+                           constant: (CGFloat)c
+{
+  NSLayoutConstraint *constraint =
+    [[NSLayoutConstraint alloc] initWithItem: view1
+                                   attribute: attr1
+                                   relatedBy: relation
+                                      toItem: view2
+                                   attribute: attr2
+                                  multiplier: mult
+                                    constant: c];
+
+  AUTORELEASE(constraint);
+  return constraint;
+}
+
++ (void) activateConstraints: (NSArray *)constraints
+{
+  NSEnumerator *en = [constraints objectEnumerator];
+  NSLayoutConstraint *c = nil;
+
+  while ((c = [en nextObject]) != nil)
+    {
+      [NSLayoutConstraint _activateConstraint: c];
+    }
+}
+
++ (void) deactivateConstraints: (NSArray *)constraints
+{
+  NSEnumerator *en = [constraints objectEnumerator];
+  NSLayoutConstraint *c = nil;
+
+  while ((c = [en nextObject]) != nil)
+    {
+      [NSLayoutConstraint _removeConstraint: c];
+    }
+}
+
+- (instancetype) initWithItem: (id)firstItem 
+                    attribute: (NSLayoutAttribute)firstAttribute 
+                    relatedBy: (NSLayoutRelation)relation 
+                       toItem: (id)secondItem
+                    attribute: (NSLayoutAttribute)secondAttribute 
+                   multiplier: (CGFloat)multiplier
+                     constant: (CGFloat)constant;
+{
+  self = [super init];
+  if (self != nil)
+    { 
+      _firstItem = firstItem;
+      _secondItem = secondItem;
+      _firstAttribute = firstAttribute;
+      _secondAttribute = secondAttribute;
+      _relation = relation;
+      _multiplier = multiplier;
+      _constant = constant;
+
+      [NSLayoutConstraint _activateConstraint: self];
+    }
+  return self;
+}
+
+// Active  
+- (BOOL) isActive
+{
+  return [activeConstraints containsObject: self];
+}
+
+- (void) setActive: (BOOL)flag
+{
+  if (flag)
+    {
+      [NSLayoutConstraint _activateConstraint: self];
+    }
+  else
+    {
+      [NSLayoutConstraint _removeConstraint: self];
+    }
+}
+
+// compare and isEqual...
+- (NSComparisonResult) compare: (NSLayoutConstraint *)constraint
+{
+  if ([self priority] < [constraint priority])
+    {
+      return NSOrderedAscending;
+    }
+  else if ([self priority] > [constraint priority])
+    {
+      return NSOrderedDescending;
+    }
+  
+  return NSOrderedSame;
+}
+
+- (BOOL) isEqual: (NSLayoutConstraint *)constraint
+{
+  BOOL result = NO;
+
+  if (YES == [super isEqual: constraint])
```

Comment:
The normal way to include the super call here is to check whether that return NO and in that case to return that value. Your implementation looks wrong to me.

Reply:
Okay, I will take another look.

Reply:
Pushed a change, please take a look.